src/rt-app_taskgroups.c

```diff
@@ -253,6 +268,60 @@ static int cgroup_mkdir(const char *name, int *offset)
 		} else if (*offset == -1) {
 			*offset = y;
 		}
+
+		if (ctrl.version == 2) {
+			size_t path_size = strlen(path) + strlen(SUBTREE_CONTROL) + 2;
+			char* cgroup_path = malloc(path_size);
+			FILE *cgroup_file;
+			snprintf(cgroup_path, path_size, "%s/%s", path, SUBTREE_CONTROL);	
+			cgroup_file = fopen(cgroup_path, "we");
+			if (!cgroup_file) {
+				perror("fopen");
+				goto error;
+			}
+			fprintf(cgroup_file, "+cpu");
+			if (fclose(cgroup_file)) {
+				perror("fclose");
+				goto error;
+			}
+			free(cgroup_path);
+			
+			path_size = strlen(path) + strlen(TYPE) + 2;
+			cgroup_path = malloc(path_size);
+			snprintf(cgroup_path, path_size, "%s/%s", path, TYPE);
+			cgroup_file = fopen(cgroup_path, "we");
+			if (!cgroup_file) {
+				perror("fopen");
+				goto error;
+			}	
+			if (fprintf(cgroup_file, "threaded") < 0) {
```

Comment:
Does this work for you?

It may but it may also break if there's "domain" controller (e.g. memory) on the ancestry path. Perhaps you can live with that, alternatively threaded mode could only be enabled on the target cgroup (provided the leaf doesn't have "domain" controller too).